javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -22,15 +24,18 @@
  * @author Ruslan Sennov
  * @since 2.0.0
  */
-public final class Vector<T> implements Kind1<Vector<?>, T>, IndexedSeq<T>, Serializable {
-
+public final class Vector<T> implements Kind1<Vector<?>, T>, IndexedSeq<T> {
```

Comment:
Oohhh, Vector.initOption() currently has return type `Option<? extends Vector<T>>`. That is not correct. Our final collections do have `? extends` any more. Could you please correct it @paplorinc?